doc/developers/developing_scikit.rst

```diff
@@ -0,0 +1,671 @@
+.. _developing_scikit:
```

Comment:
We're going to need to give this a better filename. I'm not sure yet what's right. `coding`? `implementation`? `api`?

Reply:
Personally I like 'developing' more than 'coding', but I will rely on your judgement.

Reply:
"Developing" wrt an OSS project often includes project & issue management, personnel, etc. So it covers "contributing" as well as this. Rather this section is a guide to engineering, etc.

Reply:
Grasping at straws now, but maybe use 'enhancement' or 'enhancing_scikit'. That word is used heavily at my day job :)

Reply:
This is not about enhancing. This is about how to implement/develop/code/build your own estimators. The filename should be a single word, not two. I'm okay with you choosing one of those words. Thanks